### PR TITLE
Added option to specify .gitleaksignore path

### LINKED ISSUE #1178: .gitleaksignore is not considered  when scanning individual files
**Describe the bug**
When scanning individual files, gitleaks does not search for a .gitleaksignore file in same folder, neither provides a way to specify the path to one.

This breaks the function of some git hook approaches where we want to scan only the changed files. And if there's a false positive, there's no way to flag it.

**To Reproduce**
Create a folder containing false possitves, with client ids, for example.
Create a .gitleaksignore file in the same folder and add the ignore line

Run `gitleaks detect --source <path to file> --no-git`

It will flag the leak!

**Expected behavior**
I would expect it to pick a .gitleaksignore file in the same folder of scanned file

**Basic Info (please complete the following information):**
 - OS: Ubuntu 20.04
 - Gitleaks Version: 8.16.3

cc @zricethezav

### PR DESCRIPTION
**Pull request by pacorreia**
### Description:
Added a new option for detect/protect actions to allow specify a -gitleaksignore file path or folder containing it

### Checklist:

* [x] Does your PR pass tests?
* [ ] Have you written new tests for your changes?
* [x] Have you lint your code locally prior to submission?

This PR fixes #1178

### CODE REVIEW

File: cmd/detect.go [hunk 1]
```diff
@@ -81,6 +81,23 @@ func runDetect(cmd *cobra.Command, args []string) {
 		log.Fatal().Err(err).Msg("")
 	}
 
+	gitleaksPath, err:= cmd.Flags().GetString("gitleaks-ignore-path")
```
Comment (zricethezav) [inline cmd/detect.go:84]: @pacorreia could we rename this variable to `gitleaksIgnorePath`?

File: cmd/protect.go [hunk 2]
```diff
@@ -74,6 +75,23 @@ func runProtect(cmd *cobra.Command, args []string) {
 		log.Fatal().Err(err).Msg("")
 	}
 
+	gitleaksPath, err:= cmd.Flags().GetString("gitleaks-ignore-path")
```
Comment (zricethezav) [inline cmd/protect.go:78]: @pacorreia could we rename this variable to `gitleaksIgnorePath`?

File: cmd/detect.go (hunk 1 above)
Comment (pacorreia) [reply cmd/detect.go]: Done

File: cmd/protect.go (hunk 2 above)
Comment (pacorreia) [reply cmd/protect.go]: Done

**Comment by alterEgo123** [comment]
When will this be released? I'm having an issue when I try to mount .gitleaksignore as a file use the docker image

**Pull request merged**